gointelowl/analyzer.go

```diff
@@ -0,0 +1,91 @@
+package gointelowl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+)
+
+type AnalyzerConfig struct {
+	BaseConfigurationType
+	Type                  string   `json:"type"`
+	ExternalService       bool     `json:"external_service"`
+	LeaksInfo             bool     `json:"leaks_info"`
+	DockerBased           bool     `json:"docker_based"`
+	RunHash               bool     `json:"run_hash"`
+	RunHashType           string   `json:"run_hash_type"`
+	SupportedFiletypes    []string `json:"supported_filetypes"`
+	NotSupportedFiletypes []string `json:"not_supported_filetypes"`
+	ObservableSupported   []string `json:"observable_supported"`
+}
+
+type AnalyzerService struct {
+	client *IntelOwlClient
+}
+
+/*
+* Desc: Getting the Analyzer Configurations
+* Endpoint: GET "/api/get_analyzer_configs"
+ */
+func (analyzerService *AnalyzerService) GetConfigs(ctx context.Context) (*[]AnalyzerConfig, error) {
+	requestUrl := fmt.Sprintf("%s/api/get_analyzer_configs", analyzerService.client.options.Url)
+	request, err := http.NewRequest("GET", requestUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	successResp, err := analyzerService.client.newRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	analyzerConfigurationResponse := map[string]AnalyzerConfig{}
+	if unmarshalError := json.Unmarshal(successResp.Data, &analyzerConfigurationResponse); unmarshalError != nil {
+		return nil, unmarshalError
+	}
+
+	analyzerNames := make([]string, 0)
+	// *getting all the analyzer key names!
+	for analyzerName := range analyzerConfigurationResponse {
+		analyzerNames = append(analyzerNames, analyzerName)
+	}
+	// * sorting them alphabetically
+	sort.Strings(analyzerNames)
+	analyzerConfigurationList := []AnalyzerConfig{}
+	for _, analyzerName := range analyzerNames {
+		analyzerConfig := analyzerConfigurationResponse[analyzerName]
+		analyzerConfigurationList = append(analyzerConfigurationList, analyzerConfig)
+	}
+	return &analyzerConfigurationList, nil
+}
+
+/*
+* Desc: Getting the Analyzer Configurations
+* Endpoint: GET "/api/analyzer/{NameOfAnalyzer}/healthcheck"
+ */
+func (analyzerService *AnalyzerService) HealthCheck(ctx context.Context, analyzerName string) (bool, error) {
+	requestUrl := fmt.Sprintf("%s/api/analyzer/%s/healthcheck", analyzerService.client.options.Url, analyzerName)
+	request, err := http.NewRequest("GET", requestUrl, nil)
+	if err != nil {
+		return false, err
+	}
+	status := StatusResponse{}
+	successResp, err := analyzerService.client.newRequest(ctx, request)
+	if err != nil {
+		return false, err
+	}
+	if unmarshalError := json.Unmarshal(successResp.Data, &status); unmarshalError != nil {
+		return false, unmarshalError
+	}
+	return status.Status, nil
+}
+
+func (analyzerConfig *AnalyzerConfig) Display() error {
+	jsonBytes, err := json.Marshal(analyzerConfig)
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(jsonBytes))
+	return nil
```

Comment:
Is this method needed? I pointed this out in tag service pr as well!

Reply:
Well in my opinion a custom Display method should be applied so we can easily pretty print the data. @sp35 and @eshaan7 what do you guys think?

Reply:
It is not a requirement as we are making a Library not a CLI!